servlet/src/main/java/com/linecorp/armeria/server/servlet/DefaultServletService.java

```diff
@@ -0,0 +1,124 @@
+ * under the License.
+ */
+package com.linecorp.armeria.server.servlet;
+
+import static java.util.Objects.requireNonNull;
+
+import javax.servlet.http.HttpServlet;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.common.AggregatedHttpRequest;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpResponseWriter;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * An {@link HttpService} which handles {@link HttpRequest} and forward to Servlet APIs,
+ * and write {@link HttpResponse} to client.
+ */
+public class DefaultServletService implements HttpService {
+    private static final Logger logger = LoggerFactory.getLogger(DefaultServletService.class);
+    private final DefaultServletContext servletContext;
+    private ServletHttpRequest request;
+    private ServletHttpResponse response;
+    private ServiceRequestContext serviceRequestContext;
+
+    /**
+     * Get http servlet request.
+     */
+    public ServletHttpRequest getRequest() {
+        return request;
+    }
+
+    /**
+     * Get servlet context.
+     */
+    public DefaultServletContext getServletContext() {
+        return servletContext;
+    }
+
+    /**
+     * Get http servlet response.
+     */
+    public ServletHttpResponse getResponse() {
+        return response;
+    }
+
+    /**
+     * Get channel handler context.
+     */
+    public ServiceRequestContext getServiceRequestContext() {
+        return serviceRequestContext;
+    }
+
+    /**
+     * A class which helps a {@link DefaultServletService} have a {@link HttpServlet}.
+     */
+    public DefaultServletService(DefaultServletContext servletContext) {
+        requireNonNull(servletContext, "servletContext");
+        this.servletContext = servletContext;
+    }
+
+    /**
+     * handles {@link HttpRequest} and forward to Servlet APIs.
+     */
+    @Override
+    public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exception {
+        requireNonNull(ctx, "ctx");
+        requireNonNull(req, "req");
+        serviceRequestContext = ctx;
+        final HttpResponseWriter res = HttpResponse.streaming();
+        req.aggregate().handle((aReq, cause) -> {
+            if (cause != null) {
+                logger.warn("{} Failed to aggregate a request:", ctx, cause);
+                if (res.tryWrite(ResponseHeaders.of(HttpStatus.INTERNAL_SERVER_ERROR))) {
+                    res.close();
+                }
+                return null;
+            }
+            process(res, aReq);
+            return null;
+        });
+        return res;
+    }
+
+    private void process(HttpResponseWriter res, AggregatedHttpRequest req) {
```

Comment:
Because we are just handling sync servlet (we are going to handle the async servlets later) we must use `blockingTaskExecutor`.
```java
ctx.blockingTaskExecutor().execute(() -> {
    ...
});
// or we just can do it in handle.
req.aggregate().handleAsync((aReq, cause) -> {
    ...
}, ctx.blockingTaskExecutor());
```

Reply:
Agree, I will add ctx.blockingTaskExecutor() .